src/Sylius/Component/Core/Provider/LastYearSalesSummaryByMySQLProvider.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Provider;
+
+use Doctrine\ORM\EntityRepository;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+
+final class LastYearSalesSummaryByMySQLProvider implements LastYearSalesSummaryProviderInterface
+{
+    /** @var OrderRepositoryInterface|EntityRepository */
+    private $orderRepository;
+
+    public function __construct(OrderRepositoryInterface $orderRepository)
+    {
+        $this->orderRepository = $orderRepository;
+    }
+
+    /** @psalm-suppress PossiblyUndefinedMethod */
```

Comment:
Is it needed for the whole method? It'd be safer to add it to a single expression instead.